src/perf/best-practices.md

```diff
@@ -0,0 +1,425 @@
+  by many of the widget classes.
+
+---
+
+## Implement grids and lists thoughtfully 
+
+How your grids and lists are implemented
+might be causing performance problems for your app.
+This section describes an important best
+practice when creating grids and lists,
+and how to determine whether your app uses
+excessive layout passes.
+
+### Be lazy!
+
+When building a large grid or list,
+use the lazy builder methods, with callbacks.
+That ensures that only the visible portion of the
+screen is built at startup time.
+
+For more information and examples, check out:
+
+* [Working with long lists][] in the [Cookbook][]
+* [Creating a `ListView` that loads one page at a time][]
+  a community article by AbdulRahman AlHamali
+* [`Listview.builder`][] API
+
+[Cookbook]: {{site.url}}/cookbook
+[Creating a `ListView` that loads one page at a time]: {{site.medium}}/saugo360/flutter-creating-a-listview-that-loads-one-page-at-a-time-c5c91b6fabd3
+[`Listview.builder`]: {{site.api}}/flutter/widgets/ListView/ListView.builder.html
+[Working with long lists]: {{site.url}}/cookbook/lists/long-lists
+
+### Avoid intrinsics
+
+For information on how intrinsic passes might be causing
+problems with your grids and lists, see the next section.
+
+---
+
+## Minimize layout passes caused by intrinsic operations
+
+If you've done much Flutter programming, you are
+probably familiar with [how layout and constraints work][]
+when creating your UI. You might even have memorized Flutter's
+basic layout rule: **Constraints go down. Sizes go up.
+Parent sets position.**
+
+For some widgets, particularly grids and lists,
+the layout process can be expensive.
+Flutter strives to perform just one layout pass
+over the widgets but, sometimes,
+a second pass (called an _intrinsic pass_) is needed,
+and that can slow performance.
+
+### What is an intrinsic pass?
+
+An intrinsic pass happens when, for example,
+you want all cells to have the size
+of the biggest or smallest cell (or some
+similar calculation that requires polling all cells).
+
+For example, consider a large grid of `Card`s.
+A grid should have uniformly sized cells,
+so the layout code performs a pass,
+starting from the root of the grid (in the widget tree),
+asking **each** card in the grid (not just the
+visible cards) to return
+its _intrinsic_ size&mdash;the size
+that the widget prefers, assuming no constraints.
+With this information,
+the framework determines a uniform cell size,
+and re-visits all grid cells a second time,
+telling each card what size to use. 
+
+### Debugging intrinsic passes
+
+To determine whether you have excessive intrinsic passes,
+enable the **[Track layouts option][]**
+in DevTools (disabled by default),
+and look at the app's [stack trace][]
+to learn how many layout passes were performed.
+{% comment %}
+TBD: From Kenzie: Tell users to look for intrinsic timeline events.
+They will possibly be named: '$runtimeType intrinsics'.
```

Comment:
That is the current name the framework gives these events.

Reply:
thx!